### PR TITLE
Feature/Upgrade to New-Style Django Middleware [PLAT-579]

### PR DESCRIPTION
**Pull request by pattisdr**
## Purpose

Upgrade MIDDLEWARE_CLASSES in favor of MIDDLEWARE, which helps with upgrading Django to 2.0, as well as Django-silk.

## Changes
- Update our current MIDDLEWARE_CLASSES to MIDDLEWARE.
- Upgrade django-cors-headers
- Upgrade django-debug-toolbar
- Fix origin_not_found_in_white_list CorsMiddleware override (method has been renamed)

## Ticket

https://openscience.atlassian.net/browse/PLAT-579

### CODE REVIEW

File: admin/base/settings/defaults.py [hunk 1]
```diff
@@ -244,7 +244,7 @@
 
 if DEBUG:
     INSTALLED_APPS += ('debug_toolbar', 'nplusone.ext.django',)
-    MIDDLEWARE_CLASSES += ('debug_toolbar.middleware.DebugToolbarMiddleware', 'nplusone.ext.django.NPlusOneMiddleware',)
+    MIDDLEWARE += ('debug_toolbar.middleware.DebugToolbarMiddleware', 'nplusone.ext.django.NPlusOneMiddleware',)
```
Comment (pattisdr) [inline admin/base/settings/defaults.py]: moving back to add'l development, might need some changes for debug toolbar in admin app

File: api/base/settings/defaults.py [hunk 2]
```diff
@@ -194,7 +194,7 @@
 # Set dynamically on app init
 ORIGINS_WHITELIST = ()
 
-MIDDLEWARE_CLASSES = (
+MIDDLEWARE = (
```
Comment (pattisdr) [inline api/base/settings/defaults.py]: Will need to update your local.py as well

**Comment by Johnetordoff** [comment]
@pattisdr How did you test the CORSMiddleware to make sure it was working?

**Comment by pattisdr** [comment]
@Johnetordoff, we have [existing tests for CORSMiddleware](https://github.com/CenterForOpenScience/osf.io/blob/develop/api_tests/base/test_middleware.py#L29), which failed when I updated django-cors-headers.  Luckily these tests were there, because it alerted me to the fact that django-cors-headers had renamed their `origin_not_found_in_white_lists` method to `origin_found_in_white_lists`, which we override.  Once I made the appropriate changes to our override, the tests passed again.

**Pull request merged**